packages/gensx-core/src/wait-for-input.ts

```diff
@@ -0,0 +1,40 @@
+import { Component } from "./component.js";
+import { getCurrentContext } from "./context.js";
+
+// TODO: Is there a security issue here? Does this endpoint need to be authenticated?
+function getCallbackUrl() {
+  return `${process.env.GENSX_API_BASE_URL}/org/${process.env.GENSX_ORG}/workflowExecutions/${process.env.GENSX_EXECUTION_ID}/runs/${process.env.GENSX_EXECUTION_RUN_ID}/resume`;
+}
+
+// TODO: Name this better
+export async function waitForInput<T extends Record<string, unknown>>(
+  trigger: (callbackUrl: string) => Promise<void>,
+  // schema: z.ZodSchema<T>, // TODO
+): Promise<T> {
+  // TODO: We should do some locking here to prevent multiple simultaneous waitForInput calls.
```

Comment:
Locking is good, but even more so we need to design a shutdown routine:

1. Do we let outstanding component trees resolve to completion? 
2. Do we kill outstanding components? 
3. Do we take a final checkpoint and reject any updates that come in after the fact? 

Answers to these questions are very tightly coupled to the guarantees we are able to provide (certain choices could move us from exactly once to at most once...)